### PR TITLE
Add an endpoint for creating pipeline tokens

### PR DESCRIPTION
**Pull request by kjsanger**
Add a POST endpoint to be used with the admin token to create tokens for a pipeline. There is no limit on the number of tokens that can be created for a particular pipeline.

### CODE REVIEW

**Comment by mgcam** [comment]
Looks good to me

File: src/npg_porch/endpoints/pipelines.py [hunk 1]
```diff
@@ -84,6 +84,32 @@ async def get_pipeline(
...
+    token_desc: str,
+    db_accessor=Depends(get_DbAccessor),
+    permissions=Depends(validate)
+) -> Response:
+    try:
+        token = await db_accessor.create_pipeline_token(name=pipeline_name, desc=token_desc)
+    except NoResultFound:
+        raise HTTPException(status_code=404,
+                            detail=f"Pipeline '{pipeline_name}' not found")
+    return Response(status_code=status.HTTP_201_CREATED, content=token, media_type="text/plain")
```
Comment (nerdstrike) [inline src/npg_porch/endpoints/pipelines.py:110]: I don't care that much, but it's a bit naughty to return different content-types depending on whether the request was executed correctly. Content negotiation would be ideal, but fastapi isn't very good at this side of API compliance.

**Comment by kjsanger** [comment]
Err.. okay - I was working on a fix. I'll open a new PR to address it. Your comment also made me consider whether having JSON Content-Type for everything apart from one call was a good idea.

**Comment by nerdstrike** [comment]
Oops. Sorry!

**Comment by kjsanger** [comment]
Although now if I return `Token` I'm faced with leaking internal database keys (`Token.token_id`), so I'll see if I can avoid that.

**Pull request merged**